### DISCUSSION TITLE
Controller defaultValue attribute expects type to be "never"

### DISCUSSION THREAD

**Discussion by alexfestbaum** [react-hook-form/react-hook-form#5525] [Q&A]
**Describe the bug**
Setting the defaultValue of a Controller component yields a typescript error of "Error: Type 'string' is not assignable to type 'undefined'.ts(2322)". This actually should be string given my useForm type definition.

**To Reproduce**

To demonstrate the behavior I forked one of the examples located in the documentation of the useFieldArray API. The original codesandbox was: https://codesandbox.io/s/calc-i231d

I forked and replaced one of the inputs by an input that is implemented with the Controller component:

https://codesandbox.io/s/calc-forked-ozfvh?file=/src/App.tsx

Upon setting the defaultValue in line 72 the mentioned error is given.

The error was first introduced with v7.6.9

**Comment by bluebill1049** [comment]
```name={`cart.${index}.name` as `cart.0.name`}```

<img width="921" alt="Screen Shot 2021-06-08 at 8 14 26 pm" src="https://user-images.githubusercontent.com/10513364/121168291-20732a00-c896-11eb-8fc4-216926510396.png">
